### ISSUE TITLE
DocumentRepo broken with some ContentProviders since 1.8.27

### ISSUE THREAD

**Issue by r4dian** [orgzly-revived/orgzly-android-revived#381]
- Settings > sync > repositories > +
- Directory > browse
- in the file picker, hit the ≡ icon, chose Nextcloud
- Enter the folder you wanna use and hit `USE THIS FOLDER` then `Allow`
- Hit the checkmark icon, and return to the Notebooks screen
- Pull down to try and sync
- No notebooks are added, a truncated error is shown (screenshot below).

- [x] I have searched for existing issues that may be the same as or related to mine.

Please complete the following information:

Device: Sony Xperia 10 III (aka XQ-BT52)
Android Version: 13
Orgzly Revived Version: 1.8.28 (via f-droid )
Sync repository type (directory/Dropbox/WebDAV/Git): directory

1. 
![image](https://github.com/user-attachments/assets/5429b001-c7c9-47dc-89ad-4b2eca7c33b8)

2. 
![image](https://github.com/user-attachments/assets/022d870c-ee46-40e0-a790-9820caa854fe)

**Comment by r4dian** [comment]
Yep, this is how I had been using Nextcloud, since I had the NC app anyway.

I mentioned in my other issue & should have copied here that I missed several updates cos I didn't know f-droid had newer versions than one I was on, so unfortunately don't know in which version this broke.

**Comment by amberin** [comment]
If you can test with 1.8.26, that would be great. (It's still on Fdroid.)

Also, take a look at [this discussion](https://github.com/orgzly-revived/orgzly-android-revived/discussions/304#discussioncomment-10576780) and see if the same problems apply to your setup, once it works again. Of it does, you may be better off adding your Nextcloud folder as a WebDAV repository.

**Comment by r4dian** [comment]
Hello again. Yes, it works with 1.8.26 so that can get me by for now.

**Comment by amberin** [comment]
That's interesting, sounds like it could be the same behavior as what I describe in that discussion thread: the server reports different timestamps a while after changes have been synced to the webdav server. But it doesn't seem to bother you (or @sethidden) that much, so maybe it's not a big issue?

I'll dust off my branch with ContentRepo tweaks and maybe you can help me test it?

**Comment by amberin** [comment]
It's probably the code handling subdirectories which is the culprit. The on/off toggle basically just hides subfolder content in order to avoid unwanted files, but the code is very much the same.

The handling of subfolders in the "directory" repo type is very crude and makes dangerous assumptions about what the URLs will look like. As I mentioned elsewhere, I have made some work to improve it, but abandoned it due to the modification time issues. I'll revisit that, but I'm not yet sure if the mtime behavior will be a hindrance.

**Comment by Forge36** [comment]
> It's probably the code handling subdirectories which is the culprit. The on/off toggle basically just hides subfolder content in order to avoid unwanted files, but the code is very much the same.
> 
> The handling of subfolders in the "directory" repo type is very crude and makes dangerous assumptions about what the URLs will look like. As I mentioned elsewhere, I have made some work to improve it, but abandoned it due to the modification time issues. I'll revisit that, but I'm not yet sure if the mtime behavior will be a hindrance.

I wonder if this relates to my issue?

**Comment by amberin** [comment]
I'm wondering the same thing. Clearly it's not quite the same, since your problems appeared with 1.8.29. But the 403 is interesting, it could be something about the URL not being parsed or converted in the right way.

**Comment by r4dian** [comment]
Accidentally updated to the latest on fdroid and the issue remains.

Rolled back to 1.8.26 again to continue using the app.

**Comment by amberin** [comment]
I am polishing a fix for this issue, so that we no longer try to guess file names by looking at content:// URLs.

**Comment by amberin** [comment]
@r4dian @sethidden If you could try one of the builds at the bottom of this page, that would be awesome: https://github.com/orgzly-revived/orgzly-android-revived/actions/runs/13957405133

**Comment by amberin** [comment]
@sethidden Are you perhaps confusing this issue with #218 / #536? This one is about not being able to load files in the repo.

**Comment by r4dian** [comment]
> [@r4dian](https://github.com/r4dian) [@sethidden](https://github.com/sethidden) If you could try one of the builds at the bottom of this page, that would be awesome: https://github.com/orgzly-revived/orgzly-android-revived/actions/runs/13957405133

Hello - Yes, my original issue with file's owned by the Nextcloud app is fixed in the F-Droid apk on the link you shared.
